rag-toolchain/src/stores/postgres_vector_store.rs

```diff
@@ -1,11 +1,14 @@
+use crate::clients::traits::AsyncEmbeddingClient;
 use crate::common::embedding_shared::EmbeddingModel;
 use crate::common::types::{Chunk, Embedding};
+use crate::retrievers::postgres_vector_retriever::PostgresVectorRetriever;
 use crate::stores::traits::EmbeddingStore;
```

Comment:
I do wonder if instead of trying to connect when we initialise the database object it would be better to use type state for connected and not connected